riot/runtime/scheduler/scheduler.ml

```diff
@@ -26,12 +27,15 @@ type io = {
   mutable calls_send : int;
 }
 
+type blocking = { scheduler : t; domain : unit Domain.t }
```

Comment:
Potentially we don't need this type but when you said

> new Blocking_scheduler that wraps a Scheduler and adds logic for removing the domains from the blocking_processes domain list

I'm not 100% sure in OCaml what the best way of having a "wrapping" type is without having say a Scheduler trait like in Rust i.e. adhoc polymorphism. Is this type + the `Blocking_scheduler` a good way of doing this do you think?

(I'm actually not sure we technically need to keep a list as the domain lives as long as the Scheduler's loop, so if we shut the scheduler down the Domain should be closed/freed/whatever you call it automatically. But it's probably still nice to have a typed record for this and keep that data around in case we need to add to it later)